src/artifacts.js

```diff
@@ -261,16 +378,55 @@ api.declare({
     });
   }
 
-  // Publish message about artifact creation
-  await this.publisher.artifactCreated({
-    status:         task.status(),
-    artifact:       artifact.json(),
-    workerGroup,
-    workerId,
-    runId,
-  }, task.routes);
+
+  // This event is *invalid* for s3/azure storage types so we'll stop sending it.
+  // It's only valid for error, reference and blob, but we should only send it
+  // here for error and reference storageTypes
+  if (artifact.storageType === 'error' || artifact.storageType === 'reference') {
+    // Publish message about artifact creation
+    await this.publisher.artifactCreated({
+      status:         task.status(),
+      artifact:       artifact.json(),
+      workerGroup,
+      workerId,
+      runId,
+    }, task.routes);
+  }
 
   switch (artifact.storageType) {
+    case 'blob':
+      let requests;
+      let uploadId;
+      // If we're supposed to do a multipart upload, we should generate an UploadId
+      // if it doesn't already exist.  We should store that ID in the entity
+      if (artifact.details.parts) {
+        requests = await this.s3Controller.generateMultipartRequests({
+          bucket: artifact.details.bucket,
+          key: artifact.details.bucket,
+          uploadId: uploadId,
+          parts: artifact.details.parts,
+        });
+      } else {
+        let singlePartRequest = await this.s3Controller.generateSinglepartRequests({
+          bucket: artifact.details.bucket,
+          key: artifact.details.key,
+          sha256: artifact.details.sha256,
+          size: artifact.details.size,
+          metadata: {taskId, runId, name},
+          tags: {taskId, runId, name},
+          contentType: artifact.contentType,
+          contentEncoding: artifact.details.encoding || undefined,
+        });
+        requests = [singlePartRequest];
+      }
+      res.reply({
+        storageType: 'blob',
+        contentType: artifact.contentType,
+        contentLength: artifact.size,
+        contentSHA256: artifact.sha256,
+        requests: requests,
```

Comment:
nope, because requests is a list of requests.  if we were returning information about parts, and that list was a list of parts, then we'd call it parts.